.github/workflows/test-wheel-linux.yml

```diff
@@ -3,32 +3,137 @@ name: "CI: Test wheels"
 on:
   workflow_call:
     inputs:
-      host-platform:
+      build-type:
         type: string
         required: true
-      python-version:
+      host-platform:
         type: string
         required: true
       build-ctk-ver:
         type: string
         required: true
-      cuda-version:
-        type: string
-        required: true
-      local-ctk:
-        type: string
-        required: true
-      runner:
+      matrix_filter:
         type: string
-        required: true
+        default: "."
 
 jobs:
+  compute-matrix:
+    runs-on: ubuntu-latest
+    env:
+      BUILD_TYPE: ${{ inputs.build-type }}
+      ARCH: ${{ (inputs.host-platform == 'linux-64' && 'amd64') ||
+                (inputs.host-platform == 'linux-aarch64' && 'arm64') }}
+    outputs:
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+    steps:
+      - name: Validate Test Type
+        run: |
+          if [[ "$BUILD_TYPE" != "pull-request" ]] && [[ "$BUILD_TYPE" != "nightly" ]] && [[ "$BUILD_TYPE" != "branch" ]]; then
+              echo "Invalid build type! Must be one of 'nightly', 'pull-request', or 'branch'."
+              exit 1
+          fi
+      - name: Compute Python Test Matrix
+        id: compute-matrix
+        run: |
+          set -eo pipefail
+
+          # Set a default GPU based upon architecture.
+          gpu="l4"
+          if [[ "${ARCH}" == "arm64" ]]; then
+            gpu="a100"
+          fi
+          # Add a special entry for the H100 runner on amd64.
+          special_runner=""
+          if [[ "${ARCH}" == "amd64" ]]; then
+            special_runner="- { ARCH: ${ARCH}, PY_VER: '3.13', CUDA_VER: '12.8.0', LOCAL_CTK: '1', GPU: 'H100', DRIVER: 'latest' }"
+          fi
+
+          # Please keep the matrices sorted in ascending order by the following:
+          #
+          #     [PY_VER, CUDA_VER, LOCAL_CTK, GPU, DRIVER]
+          #
+          export MATRICES="
```

Comment:
We don't really have much interaction between Python version and driver version, where we don't need the cross product of them, just coverage of each. Similarly, we don't really have much interaction between Python version and CTK install type, where we don't need the cross product of them, just coverage of each.

I think the coverage matrix we want to hit is `CUDA_VER` x `LOCAL_CTK` x `DRIVER` which would give us 12x jobs per `ARCH` today that we can spread the `PY_VER` to have coverage. When we move to 13.x, we'll then have 4 `CUDA_VER` we want to test as opposed to 3 today, which would give us 24x jobs per ARCH which I believe can still be handled reasonably easily.

Reply:
This is a great point - I wanted to get opinions. This is a literal copy of what was happening but I thought it would be great to cut it down a fair bit. It sounds like you are onboard. I figured old and new driver to have some coverage, some local versus wheel. I would gladly reduce it, I was also thinking that nightly might do more, and PRs could do less.

Reply:
Not sure I hit exactly what you were thinking, but I got it to 12 per architecture aside from the special one for `amd64` and have a mixture to cover different elements of the matrix. This could be tweaked up or down for sure or mixed with some other variations (part of why I wanted to structure it this way).

Reply:
I was thinking to make sure we have coverage on each combination of `CUDA_VER: {'11.8.0', '12.0.1', '12.8.0'}` x `LOCAL_CTK: {'0', '1'}` x `DRIVER: {'earliest', 'latest'}`

Your update looks like the `DRIVER: 'earliest'` is only covered with `LOCAL_CTK: '0'` and `CUDA_VER='11.8.0'` currently. 

Unless I'm misunderstanding and we need to cover something like a third driver version as well where it's the earliest 12.x supported version for example.

Reply:
Ah, there was no existing support for changing the driver - it wasn't clear what we wanted to cover there and I figured adding something. I proved  `earliest` does not work with CUDA `12.8.0` at all in an earlier run, so that can't be added. I get what you mean, cover as much of `CUDA_VER`, `LOCAL_CTK` and `DRIVER` as possible spread across Python versions as you see little to no interaction there. I can tweak but I won't add `earliest` back with CUDA `12.8.0` as it fails at runtime.

Reply:
> I proved `earliest` does not work with CUDA `12.8.0` at all in an earlier run, so that can't be added.

Yea, this is expected since `earliest` is a CUDA 11 driver that does not support CUDA 12. Ideally we would have a test that uses the CUDA 12.0 driver with our CUDA 12.8 build.

Doesn't need to be covered in this PR, but we should probably support something like `cuda11_earliest`, `cuda11_latest`, `cuda12_earliest`, `cuda12_latest` for the driver versions.

> Additionally CUDA 12.0.1 does not work for LOCAL_CTK set to 0.

This is a bit surprising. Any idea why?

Reply:
> > > Additionally CUDA 12.0.1 does not work for LOCAL_CTK set to 0.
> > 
> > 
> > This is a bit surprising. Any idea why?
> 
> This [line here](https://github.com/NVIDIA/cuda-python/pull/555/files#diff-bc668a2c9f2299cef15b222055b4b4d5311646caec2e7610e540cee18ae9b948L232) is where it was excluded in the original matrix.

Reason: We have no good way to express "I want the CUDA wheels associated with a certain CTK major.minor(.patch)" (https://github.com/NVIDIA/cuda-python/pull/393#discussion_r1917266401), until Emma's metapackage lands (soon, @rwgk has been testing them internally).